src/V3Subst.cpp

```diff
@@ -368,11 +400,11 @@ class SubstVisitor final : public SubstBaseVisitor {
     // CONSTRUCTORS
     explicit SubstVisitor(AstNode* nodep) { iterate(nodep); }
     virtual ~SubstVisitor() override {
-        V3Stats::addStat("Optimizations, Substituted temps", m_statSubsts);
         for (SubstVarEntry* ip : m_entryps) {
-            ip->deleteUnusedAssign();
+            m_statSubsts += ip->deleteUnusedAssign();
```

Comment:
I'm not seeing the advantage of passing the statistic up as a return value, just increment it directly in the lower function.